adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -1494,19 +1503,37 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     // contains bams, hadoop-bam is a-ok! i believe that it is better (perf) to
     // just load from a single newAPIHadoopFile call instead of a union across
     // files, so we do that whenever possible
+    def hadoopBamRead(path: String): RDD[SAMRecord] = {
+      sc.newAPIHadoopFile(path,
+        classOf[AnySAMInputFormat],
+        classOf[LongWritable],
+        classOf[SAMRecordWritable],
+        ContextUtil.getConfiguration(job)).map(_._2.get)
+    }
+    val readerFn = if (sc.hadoopConfiguration.getBoolean(ADAMContext.USE_SPARK_BAM, false)) {
+      def sparkBamRead(path: String): RDD[SAMRecord] = {
+        if (path.endsWith(".bam")) {
+          sc.loadBam(HLPath(path))
```

Comment:
I probably didn't make this clear but `sc.loadReads` is the intended main entry-point to spark-bam, and will handle `.sam`, `.bam`, and `.cram` (the latter by delegating straight to hadoop-bam; for `.sam`, it's a new implementation that is probably equivalens to hadoop-bam's… i.e. it's not designed to be faster/better for any reason, like the `.bam` support is).